parachains/runtimes/collectives/collectives-polkadot/src/impls.rs

```diff
@@ -33,36 +35,33 @@ type NegativeImbalanceOf<T, I> = <<T as pallet_alliance::Config<I>>::Currency as
 	<T as frame_system::Config>::AccountId,
 >>::NegativeImbalance;
 
-type CurrencyOf<T, I> = <T as pallet_alliance::Config<I>>::Currency;
-
 type BalanceOf<T, I> = <<T as pallet_alliance::Config<I>>::Currency as Currency<
 	<T as frame_system::Config>::AccountId,
 >>::Balance;
 
 /// Implements `OnUnbalanced::on_unbalanced` to teleport slashed assets to relay chain treasury account.
-pub struct ToParentTreasury<TreasuryAcc, TempAcc, T, I = ()>(
-	PhantomData<(TreasuryAcc, TempAcc, T, I)>,
-);
+pub struct ToParentTreasury<T, I = ()>(PhantomData<(T, I)>);
 
-impl<TreasuryAcc, TempAcc, T, I: 'static> OnUnbalanced<NegativeImbalanceOf<T, I>>
-	for ToParentTreasury<TreasuryAcc, TempAcc, T, I>
+impl<T, I: 'static> OnUnbalanced<NegativeImbalanceOf<T, I>> for ToParentTreasury<T, I>
 where
-	TreasuryAcc: Get<AccountIdOf<T>>,
-	TempAcc: Get<AccountIdOf<T>>,
-	T: pallet_xcm::Config + frame_system::Config + pallet_alliance::Config<I>,
-	[u8; 32]: From<AccountIdOf<T>>,
+	T: pallet_alliance::Config<I> + frame_system::Config,
+	[u8; 32]: From<AccountId>,
 	BalanceOf<T, I>: Into<Fungibility>,
-	<<T as frame_system::Config>::RuntimeOrigin as OriginTrait>::AccountId: From<AccountIdOf<T>>,
+	<T as frame_system::Config>::AccountId: From<AccountId>,
 {
 	fn on_unbalanced(amount: NegativeImbalanceOf<T, I>) {
-		let temp_account: AccountIdOf<T> = TempAcc::get();
-		let treasury_acc: AccountIdOf<T> = TreasuryAcc::get();
+		let amount = match amount.drop_zero() {
```

Comment:
submission deposit for ambassador referendum is zero, we do not teleport 0 amount

Reply:
the rest changed of the impl just simplification